parflow_data_management/transport/migrations/0001_initial.py

```diff
@@ -0,0 +1,48 @@
+# Generated by Django 3.0.10 on 2020-10-30 20:36
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+import django_extensions.db.fields
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='KeyPair',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created', django_extensions.db.fields.CreationDateTimeField(auto_now_add=True, verbose_name='created')),
+                ('modified', django_extensions.db.fields.ModificationDateTimeField(auto_now=True, verbose_name='modified')),
+                ('name', models.CharField(blank=True, max_length=255, verbose_name='Name of Key Pair')),
+                ('public_key', models.CharField(max_length=4096, verbose_name='Public key for ssh connection')),
+                ('_private_key_file', models.FilePathField()),
+                ('owner', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='key_pairs', to=settings.AUTH_USER_MODEL)),
+            ],
+            options={
+                'get_latest_by': 'modified',
+                'abstract': False,
+            },
+        ),
+        migrations.CreateModel(
+            name='SSHConnectionInfo',
+            fields=[
+                ('id', models.AutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('created', django_extensions.db.fields.CreationDateTimeField(auto_now_add=True, verbose_name='created')),
+                ('modified', django_extensions.db.fields.ModificationDateTimeField(auto_now=True, verbose_name='modified')),
+                ('key_pair', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='ssh_connections', to='transport.KeyPair')),
+                ('user', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='ssh_connections', to=settings.AUTH_USER_MODEL)),
```

Comment:
I am not sure SSHConnectionInfo really models anything valuable? The ssh key pair already has a user associated with is?

Reply:
The idea was to separate out the information stored in `config.ssh` in the cumulus cluster model. https://github.com/Kitware/cumulus/blob/bfb61121e76d42f56814d7fa02a90670d4854715/tests/cases/transport_test.py#L51

However, we won't be storing a password and could easily move the other fields (hostname, key_pair, etc.) over to the cluster model, if you think that would work better.

Reply:
Added a hostname field in 68722e91c673f55633acd72c356defd7e4651bd4. Clusters can access associated public keys via the changes in 8700a67d652a6a781cd84e20941ec69ad640535d